### PR TITLE
Add DataFeeder

### LINKED ISSUE #6101: DataFeeder for fluid package
In v2, a Paddle program can read data directly from `data_reader`. We need the same feature for fluid package.

### PR DESCRIPTION
**Pull request by reyoung**
A v2 API like data feeder for book demos.
We can feed data directly from reader.

Fix #6101

### CODE REVIEW

File: python/paddle/v2/fluid/data_feeder.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+from __future__ import print_function
+from framework import Variable
+import core
+import layers
+import numpy
+import six.moves as six
+
+
+class DataToLoDTensorConverter(object):
+    def __init__(self, place, lod_level, shape, batch_size_dim, dtype):
```
Comment (QiJune) [inline python/paddle/v2/fluid/data_feeder.py]: It seems that `batch_size_dim` here is no use.

File: python/paddle/v2/fluid/data_feeder.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        self.shape = shape
+        self.batch_size_dim = batch_size_dim
+        if dtype == core.DataType.FP32:
+            self.dtype = 'float32'
+        elif dtype == core.DataType.INT64:
+            self.dtype = 'int64'
+        elif dtype == core.DataType.FP64:
+            self.dtype = 'float64'
+        elif dtype == core.DataType.INT32:
+            self.dtype = 'int32'
```
Comment (QiJune) [inline python/paddle/v2/fluid/data_feeder.py]:
```
else:
  throw exception
```

File: python/paddle/v2/fluid/data_feeder.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            cur_lod_len = len(data)
+            lod[-1].append(lod[-1][-1] + cur_lod_len)
+            for each_data in data:
+                self._feed_impl_(each_data, lod[:-1], lod_level - 1)
+
+    def done(self):
+        arr = numpy.array(self.data, dtype=self.dtype).reshape(self.shape)
+        t = core.LoDTensor()
+        t.set(arr, self.place)
+        if self.lod_level != 0:
```
Comment (QiJune) [inline python/paddle/v2/fluid/data_feeder.py]: self.lod_level > 0

File: python/paddle/v2/fluid/data_feeder.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        self.feed_shapes = []
+        self.feed_lod_level = []
+        for each_var in feed_list:
+            if not isinstance(each_var, Variable):
+                raise TypeError("Feed list should contain a list of variable")
+            self.feed_dtypes.append(each_var.dtype)
+            self.feed_names.append(each_var.name)
+            shape = each_var.shape
+            batch_size_dim = -1
+            for i, s in enumerate(shape):
```
Comment (QiJune) [inline python/paddle/v2/fluid/data_feeder.py]: Is not batch_size_dim always in the first dimension?

File: python/paddle/v2/fluid/data_feeder.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            self.feed_lod_level.append(each_var.lod_level)
+            self.feed_shapes.append((batch_size_dim, shape))
+
+        self.place = place
+
+    def feed(self, iterable):
+        converter = []
+        for lod_level, shape, dtype in six.zip(
+                self.feed_lod_level, self.feed_shapes, self.feed_dtypes):
+            batch_size_dim, shape = shape
```
Comment (QiJune) [inline python/paddle/v2/fluid/data_feeder.py]: This line take batch_size_dim as the first dimension.

File: python/paddle/v2/fluid/data_feeder.py (hunk 4 above)
Comment (reyoung) [reply python/paddle/v2/fluid/data_feeder.py]: No, batch_size_dim is not the first dimension if we use static RNN

File: python/paddle/v2/fluid/data_feeder.py (hunk 5 above)
Comment (reyoung) [reply python/paddle/v2/fluid/data_feeder.py]: No. shape is a tuple before.

File: python/paddle/v2/fluid/data_feeder.py (hunk 1 above)
Comment (reyoung) [reply python/paddle/v2/fluid/data_feeder.py]: Yes. batch_size_dim could be removed.

**Pull request merged**